### ISSUE TITLE
[5.4] Middleware can't get request uri and attributes

### ISSUE THREAD

**Issue by matejvelikonja** [laravel/lumen-framework#556]
After upgrading to Lumen 5.4 I found this issue. Issue occurs only in tests.

When running tests, request does not get URI, attributes, ... Everything is empty.

I've tried also on clean install of Lumen 5.4. And I get the same error.

I've created an example test to demonstrate this bug. https://github.com/matejvelikonja/l54-test-bug/commit/320eb227ef076af65a79ea0c17f37e6627ccb8ef

If I change downgrade this demo to Lumen 5.3, the test starts to work.

**Comment by Johnee** [comment]
+1

**Comment by Arhimandril** [comment]
If i switch up [these two lines](https://github.com/laravel/lumen-framework/blob/0524a38dc8c445d6ce984dd7bf38dd14afeafbb0/src/Concerns/RoutesRequests.php#L551-L552) then my tests are passing, otherwise the request instance seems to get clobbered. Might be a container bug.

**Comment by Arhimandril** [comment]
Yup, here's a stacktrace
```
[0] Laravel\Lumen\Application->registerRequestBindings:431
[1] Laravel\Lumen\Application->make:203
[2] Illuminate\Container\Container->rebound:478
[3] Illuminate\Container\Container->instance:359
[4] Laravel\Lumen\Application->parseIncomingRequest:552
[5] Laravel\Lumen\Application->dispatch:523
[6] Laravel\Lumen\Application->handle:458
```
Setting the request instance calls `rebound` which calls `make` which calls `registerRequestBindings` and that is where the request is clobbered in the container. Setting `ranServiceBinders['registerRequestBindings']` _before_ setting the request instance makes sure that `registerRequestBindings` method is never called.

**Comment by matejvelikonja** [comment]
> Setting ranServiceBinders['registerRequestBindings'] before setting the request instance makes sure that registerRequestBindings method is never called.

how to achieve this?

**Comment by GrahamCampbell** [comment]
Try again with the latest 5.4 dev please.

**Comment by matejvelikonja** [comment]
I can confirm that now it's fixed with the 5.4 dev branch.

**Comment by matejvelikonja** [comment]
when will the 5.4.2 be released, so I can use this fix?
